### DISCUSSION TITLE
using GoogleProvider throws an error "invalid_client"

### DISCUSSION THREAD

**Discussion by NEARworld** [nextauthjs/next-auth#9572] [Help] [answered]
`app/api/auth/[...nextauth]/route.ts`
```tsx
import NextAuth, { AuthOptions } from 'next-auth';
import GoogleProvider from 'next-auth/providers/google';
import { PrismaAdapter } from '@next-auth/prisma-adapter';

import prisma from '@/app/libs/prismadb';

export const authOptions: AuthOptions = {
  adapter: PrismaAdapter(prisma), // translator between prisma client and database driver
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID!,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
    }),
  ],
  debug: process.env.NODE_ENV === 'development',
  session: {
    strategy: 'jwt',
  },
  secret: process.env.NEXTAUTH_SECRET!,
};
```

`app/(site)/components/AuthForm.tsx`
```tsx
  const socialAction = (provider: 'google') => {
    setIsLoading(true);

    signIn(provider, { redirect: false })
      .then((res) => {
        if (res?.error) {
          toast.error('로그인에 문제가 있습니다');
        }
        if (res?.ok) {
          toast.success('로그인 완료');
        }
      })
      .finally(() => setIsLoading(false));
  };
```

<img width="1573" alt="image" src="https://github.com/nextauthjs/next-auth/assets/102969108/da86f281-5875-4492-8c00-749c211bbdc7">

I have `GOOGLE_CLIENT_ID` `GOOGLE_CLIENT_SECRET` environment variables.
I am using prisma and mongodb.
GithubProvider works but GoogleProvider does not work.
I checked CLIENT_ID and SECRET again and everything is right.

my NEXT and NEXT_AUTH packages versions:
```
"@next-auth/prisma-adapter": "^1.0.7",
"next": "14.0.4",
"next-auth": "^4.24.5",
"prisma": "^5.7.1",
```

**Comment by jinsley8** [comment, marked as answer]
I had a similar problem using `bun install` as the package manager. If I use `yarn install` it works fine.

Otherwise, try deleting `node_modules` and the lockfile and install dependencies again.

**Reply by NEARworld** [reply]
darn.. dude this works I deleted `node_modules` and `package.lock.json` then did `npm install` again it works normally now omg
you saved my day thanks 👍

**Reply by panva** [reply]
https://github.com/nextauthjs/next-auth/issues/9558#issuecomment-1879624021

**Comment by NEARworld** [comment]
but it seems some people are facing this issue so I think this is bug.
not sure what library is causing this bug tho.
maybe `next-auth`?
Something must not be installed normally first time then deleting node modules and package.lock.json helps.

**Reply by khuezy** [reply]
Is there an upstream dependency that's installing a minor/patch version that breaks authing? I had this problem too but it went away. Then my Google One Tap also broke due to the "audience" property which is no longer required.

**Comment by bra1nDump** [comment]
The root cause is openid-client - just run `npm upgrade` but deleting lock file and installing again also works.
 More here: https://github.com/nextauthjs/next-auth/discussions/9585
